integration/ipam.go

```diff
@@ -0,0 +1,118 @@
+// +build k8srequired
+
+package integration
+
+import (
+	"net"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/ipam"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	defaultCIDRMask   = 24
+	totalBitsLength   = 32
+	privateSubnetMask = 25
+	publicSubnetMask  = 25
+	networkCIDR       = "10.1.0.0/16"
+)
+
+type awsVPC struct {
+	NetworkCIDR       string
+	PrivateSubnetCIDR string
+	PublicSubnetCIDR  string
+}
+
+func newAWSVPCBlock(client AWSClient) (awsVPC, error) {
```

Comment:
I think injecting dependencies in a functional way at runtime smells a lot. I noticed this as well in the cert-operator integration tests. With the TPR migration we use different clients and now when we refactor this we have to change all interfaces of all functions and it is just not easy to change the code. I would like to have some kind of "service" implementation like we have everywhere else. We have a pattern for injecting settings and dependencies so an implementation can provide a nice and stable interface.

Reply:
mmm I don't think uniformity is a good reason for overengineering a solution or, worse, early optimizations. This code here is not a service, just logic that addresses a required functionality. IMO if we need to reuse it, we might need to provide additional functionality, or a common API, or adding an interface for the parameters... as needed, letting it evolve. If for example, we declare now the parameter as an interface and we have a single implementation, no unit tests for this code, just having the interface sitting there for the sake of it IMO that would be a good definition of waste :) WDYT?

Reply:
The interface right now is this. 

```
func newAWSVPCBlock(client AWSClient) (awsVPC, error)
```

Tomorrow it will be this. 

```
func newAWSVPCBlock(client AWSClient, crdClient CRDClient) (awsVPC, error)
```

Later it will be this.

```
func newAWSVPCBlock(client AWSClient, crdClient CRDClient, extClient ExtClient) (awsVPC, error)
```

The clients are passed along the way and all 50 function calls of all functions have to be modified. That can't be right.

Reply:
Indeed, but in this case IMO this code has few reasons to change, I can't figure out what else a function that queries the AWS API to determine network blocks might need, other than an AWS client. If it needs anything else it is probably doing more than one thing, and it should be refactored. As I see it that advice would be hidden by a struct parameter. But anyway, please bear with me on this one, I think that in this case other approaches would make things more complicated than they need to be.

Reply:
I talk about this approach in general not only about this specific function. I am working on the TPR migration and it is already more painful than it should be to fix the integration tests.

Reply:
Would be nice to get help on this. E.g. in cert-operator we would need to apply some changes. The integration tests are failing now because there is no TPR anymore. I wanted to sort this separately and would rework this a little bit. When you just want to get it done then I can take this off of my list.

Reply:
Sure, on it, will let you know how it goes